owners/src/owners_tree.js

```diff
@@ -129,21 +154,20 @@ class OwnersTree {
   /**
    * Lists of owners which have modifiers from this tree to the root.
    *
+   * @param {!string} filename file to test ownership for.
    * @param {!OWNER_MODIFIER} modifier owner modifier.
    * @return {Owner[]} list of owners.
    */
-  getModifiedOwners(modifier) {
+  getModifiedFileOwners(filename, modifier) {
     const modifiedOwners = {};
 
-    this.allRules.forEach(rule => {
-      rule.owners
-        .filter(owner => owner.modifier === modifier)
-        .forEach(owner => {
-          if (!modifiedOwners[owner.name]) {
-            modifiedOwners[owner.name] = owner;
-          }
-        });
-    });
+    this.fileOwners(filename)
+      .filter(owner => owner.modifier === modifier)
+      .forEach(owner => {
+        if (!modifiedOwners[owner.name]) {
+          modifiedOwners[owner.name] = owner;
```

Comment:
`modifiedOwners` would work nicely as a map here.

Reply:
The issue with that, or even with adding to a set, is that there may be multiple distinct `Owner` instances which refer to the same user or team, and there is no way of overloading the equality operator used for set comparison in ES6. So this creates a dict keyed by user/team name, which ensures no duplicates. For example, something like

```
return Array.from(
      new Set(
        this.fileOwners(filename)
        .filter(owner => owner.modifier === modifier)
        .forEach(modifiedOwners.add, modifiedOwners)
      )
    );
```

Would fail `owners tree > getModifiedFileOwners > does not return duplicate owners`

Reply:
A map is a dict though? I'm thinking something like
```suggestion
const modifiedOwners = new Map();
this.fileOwners(filename).forEach(owner => {
    if (owner.modifier === modifier) {
        modifiedOwners.set(owners.name, owner);    
    }
});
```

Reply:
Whoa, first time using a code suggestion. Interesting

Reply:
I can make this change. What's the benefit of Map over Objects?

Reply:
Could be helpful if you need to iterate by `owner.name` later on. More comparisons here -
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Objects_and_maps_compared

Reply:
(I thought you meant the may function)

Reply:
lol doh

Reply:
Done!